### DISCUSSION TITLE
[Enhancement] MVU Previewer

### DISCUSSION THREAD

**Discussion by kevinjpetersen** [dotnet/maui#77] [General]
## Summary
Now that the goal is to support MVU for MAUI, it would be really cool to see some sort of Previewer for MVU so you can see your changes like, similar to how XAML has the XAML Previewer.

I'd would be tremendously awesome to see your design/layout changes, live, without having to go do a build, then a deploy, which all in all could take minutes depending on hardware, just to see a single little change.

I'm not sure if this would be too huge of a challenge to make a reality, but I think it would be amazing to have! I'm down to help with this if possible!

**Comment by libin85** [comment]
Isnt hot reload a better option that a previewer?

**Reply by I4-PJ** [reply]
Definitely not. To me seems just as "great option" due to not being able to develop an offline previewer. It is so annoying proceeding to some exact page and see the design changes. Doesn't work well with templates anyway nor 3rd party libs.

**Comment by larsduewel** [comment]
MAUI with MVU + hot reload working in Visual Studio Code would be awesome. Even tho i like Visual Studio.

**Comment by libin85** [comment]
> MAUI with MVU + hot reload working in Visual Studio Code would be awesome. Even tho i like Visual Studio.

Please checkout Comet if u have not seen it yet. Its just does exactly same what you have asked for. Comet has been a playground for MAUI when it comes to MVU

**Comment by kevinjpetersen** [comment]
> Isnt hot reload a better option that a previewer?

I understand that Hot Reload can be used, altho if you want to use that logic, then why does XAML have a Previewer, when you can just use Hot Reload too?

I wouldn't say that's a valid reason not to potentially have it :)

**Comment by kevinjpetersen** [comment]
Beside that, having the ability to use a Previewer over Debugging with Hot Reload, will save you a lot of time too. Especially if you have a slow machine that can only do so much with it's CPU and RAM. When debugging, it's not like your PC gets faster doing that :)

A Previewer allows you to relax, design in peace and not have to worry about accidentally closing the debugging sessions, having to reopen it, etc. etc.

**Comment by yesserm** [comment]
I agree that an MVU Previewer is required. In debugging using Visual Studio with the iPhone remote simulator from a MAC, I think it would be easier to watch a preview than to use Hot reload, which until now had debugging restrictions with devices that use iOS.

**Comment by zprima** [comment]
To have an Previewer would be cool. Like compose and swiftui have it. Flutter has hot reload because its fast and does not need to compile the code. But any big changes to the ui hierarchy a rebuild is required. In a preview you dont need to build/hot reload anything usually. Its just the rendered component with dumy data that you pass in.
